spring-cloud-alibaba-examples/seata-example/readme.md

```diff
@@ -0,0 +1,162 @@
+
+
+## Run example
+
+Run the Main function of `account-server`, `order-service`, `storage-service` and `business-service` respectively to start the example.
+
+After starting the example, access the following two URLs via the `GET` method of HTTP to verify the scenario of calling other services in `business-service` via RestTemplate and FeignClient, respectively.
+
+```$xslt
+http://127.0.0.1:18081/seata/feign
+
+http://127.0.0.1:18081/seata/rest
+```
+
+## How to verify the success of a distributed transaction?
+
+### Check if the Xid information was passed successfully
+
+In the Controller of `account-server`, `order-service` and `storage-service` services, the first logic executed is to output the Xid information in the RootContext, and if we see that the correct Xid information is output, i.e., it changes every time and the Xid of all services in the same. If you see that the correct Xid information is output, i.e., it changes every time and the Xid of all services in the same call is the same. If you see that the Xid information in the RootContext is correct, i.e., it changes every time, and the Xid of all the services in the same call is the same, then the Xid of Seata is passed and restored properly.
+
+### Check if the data in the database is consistent
+
+In this example, we simulate a scenario where a user purchases goods, the StorageService is responsible for deducting the inventory quantity, the OrderService is responsible for saving the order, and the AccountService is responsible for deducting the user's account balance.
+
+To demonstrate the sample, we use Random.nextBoolean() in OrderService and AccountService to throw a random exception, simulating a random exception occurring during a service call.
+
+If the distributed transaction is in effect, then the following equation should hold
+
+- `Original amount of the user (1000)` = `existing amount of the user` + `unit price of the goods (2)` * `number of orders` * `number of goods per order (2)`
+
+- `Initial quantity of goods (100)` = `existing quantity of goods` + `number of orders` * `number of goods per order (2)`
+
+## Compatible with the Spring Cloud ecosystem
+
+- Service providers that provide services through `Spring MVC` can automatically restore `Seata` context when they receive an HTTP request with `Seata` information in the header.
+
+- Support for service callers to automatically pass `Seata` contexts when called via RestTemplate.
+
+- Supports automatic passing of `Seata` context when called by a service caller via FeignClient.
+
+- Supports scenarios where `SeataClient` and `Hystrix` are used together.
```

Comment:
`Hystrix` is deprecated, remove this line.

Reply:
fixed !